src/build-tree-from-flat-array/index.ts

```diff
@@ -38,6 +38,29 @@ import { BuildTreeFn } from './types';
  *   }
  * ]
  */
+
+
 export const buildTree: BuildTreeFn = (items) => {
-  throw new Error('Not Implemented');
-};
+  const map = items.reduce(
+    (acc, item) => {
+      acc[item.id] = { ...item, children: [] };
+      return acc;
+    },
+    {} as Record<number, any>,
+  );
+
+  const tree: any[] = [];
+
+  items.forEach((item) => {
+    if (
+      item.parentId !== null &&
+      item.parentId !== undefined &&
+      map[item.parentId]
+    ) {
+      map[item.parentId].children.push(map[item.id]);
+    } else {
```

Comment:
Instead of using `else`, you can `return` early from the function in the `if` statement.
This reduces nesting and prevents the code from performing unnecessary actions.